api/src/main/java/org/pmiops/workbench/utils/WorkspaceMapper.java

```diff
@@ -1,36 +1,40 @@
 package org.pmiops.workbench.utils;
 
-import com.google.common.collect.ImmutableSet;
-import java.util.List;
-import java.util.Set;
+import static org.mapstruct.NullValuePropertyMappingStrategy.*;
+
+import org.mapstruct.CollectionMappingStrategy;
 import org.mapstruct.Mapper;
 import org.mapstruct.Mapping;
-import org.pmiops.workbench.db.model.DbStorageEnums;
+import org.mapstruct.MappingTarget;
 import org.pmiops.workbench.db.model.DbWorkspace;
 import org.pmiops.workbench.firecloud.model.FirecloudWorkspace;
 import org.pmiops.workbench.model.CdrVersion;
 import org.pmiops.workbench.model.ResearchPurpose;
-import org.pmiops.workbench.model.SpecificPopulationEnum;
 import org.pmiops.workbench.model.Workspace;
 import org.pmiops.workbench.utils.mappers.CommonMappers;
 
 @Mapper(
     componentModel = "spring",
+    collectionMappingStrategy = CollectionMappingStrategy.TARGET_IMMUTABLE,
     uses = {CommonMappers.class})
 public interface WorkspaceMapper {
 
   @Mapping(target = "researchPurpose", source = "dbWorkspace")
   @Mapping(target = "etag", source = "dbWorkspace.version", qualifiedByName = "cdrVersionToEtag")
-  @Mapping(target = "dataAccessLevel", source = "dbWorkspace.dataAccessLevelEnum")
   @Mapping(target = "name", source = "dbWorkspace.name")
   @Mapping(target = "id", source = "fcWorkspace.name")
   @Mapping(target = "googleBucketName", source = "fcWorkspace.bucketName")
   @Mapping(target = "creator", source = "fcWorkspace.createdBy")
   @Mapping(target = "cdrVersionId", source = "dbWorkspace.cdrVersion")
   Workspace toApiWorkspace(DbWorkspace dbWorkspace, FirecloudWorkspace fcWorkspace);
 
-  // This method is simply merging the research purpose, which covers only a subset of the fields
-  // in the DbWorkspace source.
+  @Mapping(target = "researchPurpose", source = "dbWorkspace")
```

Comment:
It looks like we have a method to translate a `DbWorkspace` directly into a `ResearchPurpose`. Since those types are both unique, I would've expected the inspection machinery to do the needful. NBD.

Reply:
This is a bit of an odd one since there is no researchPurpose field on DbWorkspace. Rather, it needs to read several fields off of DbWorkspace to construct ResearchPurpose through one of the other mapper functions

Reply:
It's not able to figure that out on its own though so this line is necessary.